deploy to server root. fixes #8

splitbrain opened the pull request:
This deploys OpenGrok as the default web app in tomcat. All tomcat default apps, docs and examples have been removed from the webapps dir.
A redirect from /source has been added for backward compatibility.

splitbrain committed:
deploy to server root. fixes #8

This deploys OpenGrok as the default web app in tomcat. All tomcat
default apps, docs and examples have been removed from the webapps dir.
A redirect from /source has been added for backward compatibility.

tarzanek commented on Dockerfile:
```diff
@@ -36,7 +36,12 @@ ENV JRE_HOME /usr
+RUN rm -rf /usr/local/tomcat/webapps/* && \
+    cp "/opengrok/lib/source.war" "/usr/local/tomcat/webapps/ROOT.war" && \
+    mkdir "/usr/local/tomcat/webapps/source" && \
+    echo '<% response.sendRedirect("/"); %>' > "/usr/local/tomcat/webapps/source/index.jsp"
```
:-D interesting trick
only one suggestion here - opengrok generates by default everything with prefix /source/...
(there is an option to change this in indexer, check opengrok script - OPENGROK_WEBAPP_CONTEXT  - https://github.com/oracle/opengrok/blob/master/OpenGrok#L128 , do check other OPENGROK_WAR variables, too )
so I think all above should work as easy as just doing
OPENGROK_WEBAPP_CONTEXT=/ OpenGrok deploy
(there are some small changes done to .war file itself, it will repackage for special context + other env variables (e.g. special data(index+xref) dir)
same variable will need to be used for indexer,
and it will generate smaller indexes and the redirect will happen only for some of the requests
I personally however haven't used the option for some time, so I hope there won't be any bugs, but it will definitely be a cleaner way.
What do you think?

splitbrain replied:
`OPENGROK_WEBAPP_CONTEXT` was what I tried first. It made absolutely no difference. The webapp is still deployed to `/source`, maybe that's a bug in OpenGrok. 

I could not see any problem with the URLs generated by the OpenGrok web interface (they all seem to be relative to where the Webapp is deployed to). Everything seems to work just fine.

If you think the OpenGrok deploy command is important, I can certainly use that and move the deployed war file to the right place.

tarzanek replied:
that is definitely a bug, I think it works for anything else than "/" ... so I guess it would be cool to fix it in OpenGrok script ... 
and yes, you're correct, there were fixes to fix the generated xrefs, so if they work, then it's even better, great! :)
based on your tests it seems OPENGROK_WEBAPP_CONTEXT  might need to be obsoleted and deleted then ...
in such case, then let's go and merge your original changes and we definitely need a bug in upstream to clarify the current situation (which seems to be even better than what I remember last time I tested this change of webapp name :)  (which is cool :-D )  )

splitbrain committed:
Set context and use original deploy script

As discussed in https://github.com/OpenGrok/docker/pull/9#discussion_r185698782

tarzanek commented:
so if you're ok and the changes work, then let's go and merge this ...

tarzanek commented:
the main deal with OpenGrok deploy are the fixes done to webapp if you have custom data roots or paths to config files,
but that is given and set to default in docker, so people won't really need to change this (yet, and if they will, they will just have to fix deploy .... )

splitbrain commented:
Anything that needs to be done here to get this merged or is this good to go?

tarzanek commented:
let's do it ... the only thing that needs to be done is more like administration in OpenGrok and eventually getting rid of the WEBAPP variable ...

tarzanek commented:
thank you @splitbrain , let's have a look then on WEBAPP variable and let's make it obsolete if all works as you say ...

splitbrain commented:
Thanks for merging. Can you push it to docker hub please?

tarzanek commented:
I'd hope docker hub will pick it up automatically, shouldn't it ?

tarzanek commented:
it did:

Status
Actions
Tag
Created
Last Updated
 Success
latest
2 days ago
2 days ago

so I'd hope it works (didn't test it yet)

splitbrain commented:
https://hub.docker.com/r/nagui/opengrok/builds/ was last built 8 months ago. I guess I'm looking at the wrong repo? Which one is this deployed to? I guess the README needs an update?

tarzanek commented:
https://hub.docker.com/r/opengrok/docker/
is the one I tied to the repo ...

so yes, README might be obsolete, sorry for that, do you mind fixing it?

tarzanek commented:
I didn't most probably change it, since I didn't finish my tests :(
So if the opengrok/docker repo works for you, then the README should definitely be changed ...
